src/gui/speciesViewer_input.cpp

```diff
@@ -164,6 +193,85 @@ void SpeciesViewer::contextMenuRequested(QPoint pos)
         view_.resetViewMatrix();
     else if (actionMap[selectedAction] == "CopyToClipboard")
         copyViewToClipboard(true);
+    else if (actionMap[selectedAction] == "CreateStatic")
+    {
+        auto *site =
+            species_->addSite(EmpiricalFormula::formula(species_->selectedAtoms(), [](const auto &i) { return i->Z(); }));
+        site->setStaticOriginAtoms(species_->selectedAtoms());
+        setSite(site);
+        postRedisplay();
+        emit(siteCreatedAndShown());
+    }
+    else if (actionMap[selectedAction] == "SetOriginStatic")
+    {
+        site_->setStaticOriginAtoms(species_->selectedAtoms());
+        postRedisplay();
+        emit(sitesChanged());
+    }
+    else if (actionMap[selectedAction] == "SetXStatic")
+    {
+        site_->setStaticXAxisAtoms(species_->selectedAtoms());
+        postRedisplay();
+        emit(sitesChanged());
+    }
+    else if (actionMap[selectedAction] == "SetXStatic")
+    {
+        site_->setStaticYAxisAtoms(species_->selectedAtoms());
+        postRedisplay();
+        emit(sitesChanged());
+    }
+    else if (actionMap[selectedAction] == "CreateDynamicElements")
+    {
+        auto *site =
+            species_->addSite(EmpiricalFormula::formula(species_->selectedAtoms(), [](const auto &i) { return i->Z(); }));
+        site->setType(SpeciesSite::SiteType::Dynamic);
+        std::vector<Elements::Element> elements;
+        for (const auto &i : species_->selectedAtoms())
+            elements.push_back(i->Z());
+        site->setDynamicElements(elements);
+        setSite(site);
+        postRedisplay();
+        emit(siteCreatedAndShown());
+    }
+    else if (actionMap[selectedAction] == "CreateDynamicAtomTypes")
+    {
+        auto *site =
+            species_->addSite(joinStrings(species_->selectedAtoms(), " ", [](const auto &i) { return i->atomType()->name(); }));
```

Comment:
Prefer underscores to spaces for generated names in Dissolve.
```suggestion
            species_->addSite(joinStrings(species_->selectedAtoms(), "_", [](const auto &i) { return i->atomType()->name(); }));
```